### PR TITLE
Added code for pearson match

### PR DESCRIPTION
**Pull request by trestoh**
fixed roi issues, added some log statements

changed logic for subscribing to char notifs, fixed bug in EEPROM read

removed generic char subscription (fixes battery update issue)

removed some unneeded functions

### CODE REVIEW

**Review by mzieg — APPROVED** [review]

File: Common/NumericalMethods.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EnlightenMAUI.Common
+{
+
+    public class NumericalMethods
+    {
+        static public double[] derivative(double[] xArray, double[] yArray, int order)
```
Comment (mzieg) [inline Common/NumericalMethods.cs:13]: I want to say this is derivative but I'm impressed because it's co efficient.

File: EnlightenMAUI.csproj [hunk 2]
```diff
@@ -77,6 +77,7 @@
 	</ItemGroup>
 
 	<ItemGroup>
+	  <None Remove="libraries\SiG-785-OEM\Acetaminophen.json" />
```
Comment (mzieg) [inline EnlightenMAUI.csproj:80]: I guess I'm curious why this goes in an OEM folder but okay

File: Models/Library.cs [hunk 3]
```diff
@@ -7,27 +7,36 @@
...
 
+        Wavecal wavecal;
+
         public Library(string root, Spectrometer spec)
         {
             logger.debug($"instantiating Library from {root}");
             AssetManager assets = Platform.AppContext.Assets;
-            string[] assetP = assets.List(root);
+
+            string[] assetP = assets.List("libraries/SiG-785");
```
Comment (mzieg) [inline Models/Library.cs:31]: So we don't _want_ to get acetaminophen here?

File: Models/Measurement.cs [hunk 4]
```diff
@@ -447,6 +452,9 @@ public void reload(Spectrometer spec)
         dark = spec.dark;
         postProcess();
 
+        roiStart = spec is null ? 0 : (uint)spec.eeprom.ROIHorizStart;
+        roiEnd = spec is null ? 1024 : (uint)spec.eeprom.ROIHorizEnd;
```
Comment (mzieg) [inline Models/Measurement.cs:456]: I guess I'd prefer pixels() to 1024 if possible?

**Review by mzieg — APPROVED** [review]

File: EnlightenMAUI.csproj (hunk 2 above)
Comment (trestoh) [reply EnlightenMAUI.csproj:80]: to generically separate from the previous library files

File: Models/Library.cs (hunk 3 above)
Comment (trestoh) [reply Models/Library.cs:31]: I'm not sure I understand...this constructor is called using the OEM folder in the scope view model. The string literal was from when I was debugging and should/will be removed

File: Models/Measurement.cs (hunk 4 above)
Comment (trestoh) [reply Models/Measurement.cs:456]: here spec is the spectrometer, so if that's null and this is being called that's a problem, but I do agree the previous pixel number makes more sense

**Pull request merged**